src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsVocabularyAnnotation.cs

```diff
@@ -448,6 +448,23 @@ private IEdmVocabularyAnnotatable ComputeTarget()
 
                             return new UnresolvedParameter(operationImport.Operation, parameterName, this.Location);
                         }
+
+                        IEdmEntityContainerElement containerElement = container.FindEntitySetExtended(targetSegments[1])
+                                                                      ?? container.FindSingletonExtended(targetSegments[1]) as IEdmEntityContainerElement;
+
+                        IEdmEntitySet entitySet = container.FindEntitySetExtended(targetSegments[1]);
+                        IEdmSingleton singleton = container.FindSingletonExtended(targetSegments[1]);
+
+                        if (entitySet != null)
+                        {
+                            IEdmEntityType entityType = entitySet.EntityType();
+                            IEdmProperty edmProperty = entityType.DeclaredProperties.Where(p => p.Name == targetSegments[2]).FirstOrDefault();
+
+                            if (edmProperty != null)
+                            {
+                                return new EdmAnnotationsTarget(container, entitySet, edmProperty);
+                            }
+                        }

```

Comment:
only need to resolve it as AnnotationsTarget for 3 segments?

Reply:
Fixed

Reply:
can someone use an annotation target path with a single or two segments? What will happen if someone tries that?

Reply:
@ElizabethOkerio A path must start with `entity container` followed by `container element`.
2 segments are allowed as long as they follow the above convention. However we don't need `EdmTargetPath` for container elements since you can get the entity container from the container element.
1 segment will not pass validation

Reply:
so if someone does something like this? `new EdmTargetPath(container, customer)` This will work? What will not work is if someone does this? `new EdmTargetPath(container)`

Reply:
Why I'm asking is because this logic has been added to when we have 3 segments or more but a user can call this constructor with only segment or 2 or 3 or more?  or are there validations to ensure that the `EdmTargetPath` class get called when the path has 3 segments or more?